### ISSUE TITLE
UCLO bug?

### ISSUE THREAD

**Issue by shviller** [franko/luajit-lang-toolkit#41]
This snippet
```lua
local z1
local i = 1
while i <= 2 do
  local j = i
  z1 = z1 or function() return j end
  i = i + 1
end
assert(z1() == 1)
```
(adapted from https://github.com/openresty/luajit2-test-suite/blob/master/test/misc/uclo.lua) works with LuaJIT, but fails with luajit-lang-toolkit.

**Comment by franko** [comment]
Hi,

thank you for reporting this. I have fixed the problem in the branch

https://github.com/franko/luajit-lang-toolkit/tree/fix-uclo-while-bug

but I am still working to fix the same problem in the repeat loop.

**Comment by franko** [comment]
Ok now the fix is in the master branch, commits ca5fef229c0b54037288b641a4629164df284c87 and be3f8af05c497d869956553782d81d0376862f84. I have fixed the same problem for the while and the repeat loops and as far as I can tell now the generated code is the same of luajit.
